### PR TITLE
Enable building and testing against Kokkos with pthread enabled

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

**Comment by aprokop** [comment]
Why? And if we want this, do we test this?

**Comment by dalg24** [comment]
> Why? And if we want this, do we test this?

Because I don't have OpenMP with my AppleClang.  We will, eventually.  But no OS X node on Jenkins yet.

**Comment by aprokop** [comment]
I have no opinion on this.

**Comment by aprokop** [comment]
Should #187 be resurrected instead of this one, if this something we want?

**Comment by dalg24** [comment]
> Should #187 be resurrected instead of this one, if this something we want?

Good point.  I will cherry pick d7d7f6f.  I don't want to deal with testing yet.

**Comment by dalg24** [comment]
https://github.com/arborx/ArborX/pull/307#issuecomment-549436043

Do as I say not as I do...

**Comment by dalg24** [comment]
Looking at https://github.com/arborx/ArborX/wiki/Testing I am just not sure what build to add it to

**Comment by aprokop** [comment]
> Looking at [arborx/ArborX/wiki/Testing](https://github.com/arborx/ArborX/wiki/Testing) I am just not sure what build to add it to

`Clang`

**Comment by dalg24** [comment]
> > Looking at [arborx/ArborX/wiki/Testing](https://github.com/arborx/ArborX/wiki/Testing) I am just not sure what build to add it to
> 
> `Clang`

Apparently this is not an option :)
```
-- Execution Spaces:
--     Device Parallel: NONE
CMake Error at cmake/kokkos_arch.cmake:420 (MESSAGE):
  Multiple host parallel execution spaces are not allowed! Trying to enable
  execution space PTHREAD, but execution space OPENMP is already enabled.
  Remove the CMakeCache.txt file and re-configure.
Call Stack (most recent call first):
  cmake/kokkos_tribits.cmake:181 (INCLUDE)
  CMakeLists.txt:156 (KOKKOS_SETUP_BUILD_ENVIRONMENT)


-- Configuring incomplete, errors occurred!
See also "/scratch/kokkos/build/CMakeFiles/CMakeOutput.log".
```

**Comment by aprokop** [comment]
Well, this narrows it down :) Either of the two builds without OpenMP (`CUDA-9.2-NVCC-CUDA-AWARE-MPI` or `CUDA-10.0-Clang`) would work. Apple is "Clang", so just add it to `CUDA-10.0-Clang`.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -9,6 +9,14 @@ endif()
 if(Kokkos_ENABLE_CUDA)
   list(APPEND ARBORX_DEVICE_TYPES Kokkos::CudaSpace::device_type)
 endif()
+if(Kokkos_ENABLE_PTHREAD)
+  list(APPEND ARBORX_DEVICE_TYPES Kokkos::Threads::device_type)
+endif()
+
+if(NOT ARBORX_DEVICE_TYPES)
+  message(SEND_ERROR "Kokkos_DEVICES must include at least one of 'Serial', 'OpenMP', 'Cuda' and 'Pthread'!")
```
Comment (aprokop) [inline test/CMakeLists.txt:17]:
```suggestion
  message(SEND_ERROR "Kokkos_DEVICES must include at least one of 'Serial', 'OpenMP', 'Cuda' or 'Pthread'!")
```

**Pull request merged**